doit/cmd_base.py

```diff
@@ -310,30 +310,64 @@ def _load_from(cmd, namespace, cmd_list):
         return task_list, doit_config
 
 
-class ModuleTaskLoader(TaskLoader):
+class TaskLoader2(TaskLoader):
+    """Interface of task loaders with new-style API.
+
+    The default implementation assumes tasks are loaded from a namespace, mapping identifiers to
+    elements like functions (like task generators) or constants (like configuration values).
+
+    This API update separates the loading of the configuration and the loading of the actual tasks,
+    which enables additional elements to be available during task creation.
+    """
+    def __init__(self):
+        super().__init__()
+        self.namespace = None
```

Comment:
I know I am the one who suggested to remove ``NamespaceLoader``... but **now** I am thinking it makes sense to have it. Sorry.

I guess it would make it clear that this ``namespace`` attribute is not required by the API.